### PR TITLE
bump jackson2-api from 2.12.1 to 2.13.3-285.vc03c0256d517

### PR DESCRIPTION
**Pull request by car-roll**
also bumps jenkins version to 2.263.1

cc @jtnord 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by raul-arabaolaza — APPROVED** [review]

**Comment by raul-arabaolaza** [comment]
Downstream of #334

File: pom.xml [hunk 1]
```diff
@@ -104,7 +109,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>jackson2-api</artifactId>
-      <version>2.12.1</version>
+      <version>2.13.3-285.vc03c0256d517</version>
```
Comment (jtnord) [inline pom.xml]: why are we not just using a more recent bom version that contains this?
e.g. https://github.com/jenkinsci/bom/blob/807.v6d348e44c987/bom-latest/pom.xml  (newer are available)

File: pom.xml [hunk 2]
```diff
@@ -89,11 +89,16 @@
...
+        <artifactId>bom-2.263.x</artifactId>
         <version>24</version>
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>io.jenkins.plugins</groupId>
+        <artifactId>snakeyaml-api</artifactId>
+        <version>1.29.1</version>
+      </dependency>
```
Comment (jtnord) [inline pom.xml]: contained in bom version `918.vae501d2cdc99`  (and possibly older) - is there a reason we are not updating the bom to pickup the newer versions?

**Review by jtnord — COMMENTED** [review]
Should we not just update the bom version here to pickup the newer required dependencies.

Having versions defined for dependencies when we are using the bom and a bom version has those versions seems to be sub-optimal.

Additionally I am slightly converned that the `upper-bounds` enforcer rule did not fail due to the `jackson-api` version being [lower than expected](https://github.com/jenkinsci/google-compute-engine-plugin/pull/334#issuecomment-1180438693)

File: pom.xml (hunk 1 above)
Comment (car-roll) [reply pom.xml]: you're right, will clean this all up

File: src/test/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWorkTest.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public void shouldNotCleanAnyInstance() throws Exception {
 
     getWorker().doRun();
     verify(client).listInstancesWithLabel(eq(TEST_PROJECT_ID), anyMap());
-    verifyZeroInteractions(client);
+    verifyNoMoreInteractions(client);
```
Comment (car-roll) [inline src/test/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWorkTest.java:85]: `verifyZeroInteractions` is deprecated and should use `verifyNoInteractions`. However, that causes a test failure as it finds an interaction. 
Would `verifyNoMoreInteractions` be a more appropriate test?

Comment (jtnord) [reply src/test/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWorkTest.java:85]: this looks correct to me, as you are verifying there has been an interaction before and you are checking that there are no other one!

**Review by jtnord — APPROVED** [review]

**Pull request merged**